main.cpp

```diff
@@ -425,3 +429,55 @@ void enlargeImage()
     }
   }
 }
+
+void mirrorImage()
+{
+  // Keep asking for the part of the image until a valid one is chosen
+  int choice;
+  bool isValidMirror = false;
+  while (!isValidMirror)
+  {
+    cout << "select the part of the image you want to reflect: \n"
+         << "1. Reflect the right half  of the image\n"
+         << "2. Reflect the left  half of the image\n"
+         << "3. Reflect the upper half of the image\n"
+         << "4. Reflect the lower half of the image\n";
+    cin >> choice;
+
+    // Check if the value is valid, if not ask again.
+    isValidMirror = choice == 1 || choice == 2 || choice == 3 || choice == 4;
```

Comment:
Here you should do it one step instead of checking the value of choice for every number 
```suggestion
    isValidMirror = choice >= 1 &&  choice <= 4;
```

Reply:
thanks, I'll update this change.